overrides/fips.yaml

```diff
@@ -1,18 +1,23 @@
 ---
-etcd_image_tag: v3.4.13_fips.0
+k8s_image_registry: mesosphere
+
 fips:
   enabled: true
-  etcdImageTag: "{{ etcd_image_tag }}"
+
+image_replacements:
+  "k8s.gcr.io/etcd:3.4.13-0": "{{ k8s_image_registry }}/etcd:v3.4.13_fips.0"
```

Comment:
Wont we also need to replace the apiserver, scheduler, controller-manager, proxy (or bake a kubeadm init config).. It needs the build metadata, `kube-apiserver:v1.21.6_fips.0` for instance.

Reply:
@jkoelker please see this question ☝️

Reply:
Yea anywhere that we are changing the tag on the image, we'll need an override entry. From above though it appeared that only `etcd`'s tag was changed, and the others relied on the setting of the the image registry. But from looking at the repos in docker hub, and now seeing the below `kubernetes_build_metadata` tag that changes the other images, yea we should iterate them all out here explicitly. Otherwise some are set via `kubernetes_image_tag` and then just the `etcd` is set statically.

However I'm kinda wondering if we should look at having the `kubeadm` that we build in fips mode emit the correct tags it expects for reproducibility. That is if we "validate" that the `fips.0` build all work together, we can get rid of the image overriding all together.

I'll see if I can track down where / how we're building and publishing to the fips repo below, and if it trivial or not to make that happen. If its not, then I'll iterate them for now.

Reply:
Went down a few rabbit holes, but I think I found a pretty nice solution:

https://github.com/mesosphere/konvoy-image-builder/pull/161/commits/8a49551524328fc4c580cc9fe94e5b3eaf9b33fd#diff-14c84fc26cc68826bdd8c9ee2ba0b29f31537fc6fec4f03028d38b024d7b9d6cR10-R25

This lets the build tag (the `GitVersion` field from `kubeadm version`) drive the process of selecting the image tags. There is a small fixup that I do to just the `etcd` image as upstream sets a build tag on them by default (e.g. `etcd:3.4.13-0`). I am stripping the `-[0-9]+` from it before appending on the build tag from `kubeadm`.